src/vmspawn/vmspawn.c

```diff
@@ -538,6 +656,32 @@ static int run_virtual_machine(void) {
         if (!cmdline)
                 return log_oom();
 
+        /* if we are going to be starting any units with state then create our runtime dir */
+        if (arg_tpm != 0) {
+                _cleanup_free_ char *runtime_directory = NULL;
+                const char *e = getenv("RUNTIME_DIRECTORY");
+                if (e)
+                        runtime_directory = strdup(e);
+                else if (getuid() == 0)
+                        runtime_directory = strdup("/run");
+                else {
+                        r = xdg_user_runtime_dir(&runtime_directory, "");
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to find user's runtime directory: %m");
+                }
+                if (!runtime_directory)
+                        return log_oom();
+
+                /* ensure runtime_dir/systemd/vmspawn exists */
+                our_runtime_dir = path_join(runtime_directory, "systemd/vmspawn");
+                if (!our_runtime_dir)
+                        return log_oom();
+
+                r = mkdir_p(our_runtime_dir, 0755);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to create runtime directory: %m");
```

Comment:
This should be a helper function (maybe in daemon-util.h?) There's nothing vmspawn specific about this.

Reply:
I think this might be more appropriate in path-lookup.[ch]?

Reply:
daemon-util seems to be more focussed on notify sd-notify things.

Reply:
Yeah maybe that's better